js/index.js

```diff
@@ -1,8 +1,160 @@
+    symbol: "fas fa-grin-stars",
+  },
+];
 
+const textDiv = document.querySelector(".karakter_text_wrapper");
+const characters = CharacterJson;
 
 function showKarakter() {
-  var state = document.getElementsByClassName(
-    "karakter__revealer");
-    var aNode = state[0];
-    aNode.style.visibility = aNode.style.visibility === "hidden" ? "visible" : "hidden";
+  let state = document.getElementsByClassName("karakter__revealer");
+  let aNode = state[0];
+  aNode.style.display = aNode.style.display === "none" ? "block" : "none";
 }
+function showKnowledgeBook() {
+  let state = document.getElementsByClassName("knowledgeBook__revealer");
+  let aNode = state[0];
+  aNode.style.display = aNode.style.display === "none" ? "block" : "none";
+}
+
+const iconKarakter = (characters) => {
+  const characterDivID = document.querySelector("#karakter__icon_wrapper");
+  characters.forEach((character) => {
+    const characterDiv = document.createElement("div");
+    const characterButton = document.createElement("button");
+    const characterIcon = document.createElement("i");
+    const characterTitle = document.createElement("h1");
+    const characterDiscription = document.createElement("p");
+
+    characterButton.classList.add("characterBtn");
+    // characterButton.addEventListener(
+    //   "click",
+    //   (event) => showMore(character.description, textDiv) // ik kan hier het carajter id meegeven
+
+    characterButton.addEventListener("click", (event) =>
+      main(character.description, textDiv, character.id)
+    );
+    characterIcon.setAttribute("class", `${character.symbol}`);
+    characterTitle.classList.add("characterTitle_moreInfo");
+    characterTitle.innerText = character.title;
+    characterDiscription.classList.add("characterDiscription_moreInfo");
+    characterDivID.appendChild(characterDiv);
+    characterDiv.appendChild(characterButton);
+    characterButton.appendChild(characterIcon);
+  });
+};
+
+//in var opslaan hoevell keer i text aan het lopen en wlke knop
+//hoeveel keer looptext
+// clear time out triggeren al je wilt stoppen ]
+// als er een text aan het lopee is is de text van dezelede knop aan het lopen doeniks. i het een andere knp willes text
+let timer = 0;
+
+function loopText(textLoss, show) {
+  if (textLoss) {
+    if (textLoss.length > 0) {
+      show.innerHTML += textLoss.shift();
+    } else {
+      console.log(timer);
+      stopTimer(timer);
+      return false;
+    }
+    timer = setTimeout(function () {
+      loopText(textLoss, show);
+    }, 70);
+  }
+}
+
+function stopTimer() {
+  clearTimeout(timer);
+}
+let laatste = 0;
+
+function main(text, targetDiv, id) {
+  targetDiv.innerHTML = "";
+
+  if (laatste == id) {
+    console.log("ik ben if");
+    stopTimer(timer);
```

Comment:
Je bent op het goede pad door een stopTimer functie te hebben. Alleen roep je hem nog niet op het juiste moment aan. Eigenlijk wil je de timer altijd stoppen voordat je iets uitvoert om te voorkomen dat dingen door elkaar gaan lopen. De conditie waar je de vorige id checkt met de huidige is dan ook niet nodig tenzij je iets specifieks wil doen wanneer je 2x op dezelfde item klikt.

Reply:
YHHeeaaa gelukt volgens mij bedoelde je dit